libpromises/eval_context.c

```diff
@@ -1240,7 +1240,18 @@ static bool EvalContextClassPut(EvalContext *ctx, const char *ns, const char *na
         char context_copy[CF_MAXVARSIZE];
         char canonified_context[CF_MAXVARSIZE];
 
-        strcpy(canonified_context, name);
+
+        /* Redmine #7013
+         * Fix for classes names longer than CF_MAXVARSIZE. */
+        if (strlen(name) >= CF_MAXVARSIZE - 1)
```

Comment:
You don't need the `` - 1``. By using ``>=``, you're already checking the correct length.